yorkie/src/main/kotlin/dev/yorkie/document/crdt/RHTPQMap.kt

```diff
@@ -0,0 +1,161 @@
+     */
+    fun set(key: String, value: CrdtElement): CrdtElement? {
+        var removed: CrdtElement? = null
+        val queue = elementQueueMapByKey[key]
+        if (queue != null && queue.size > 0) {
+            val node = queue.peek() as RHTPQMapNode
+            if (!node.isRemoved() && node.remove(value.createdAt)) {
+                removed = node.value
+            }
+        }
+        setInternal(key, value)
+        return removed
+    }
+
+    /**
+     * `setInternal` sets the value of the given key internally.
+     */
+    private fun setInternal(key: String, value: CrdtElement) {
+        val queueMap = elementQueueMapByKey
+        if (!queueMap.contains(key)) {
+            queueMap[key] = MaxPriorityQueue()
+        }
+
+        val node = RHTPQMapNode.of(key, value)
+        val queue = queueMap[key]
+            ?: throw IllegalStateException("The MaxPriorityQueue by $key doesn't exist.")
+        queue.add(node)
+        nodeMapByCreatedAt[value.createdAt.toIDString()] = node
+    }
+
+    /**
+     * `delete` deletes deletes the Element of the given key.
+     *
+     * @param createdAt to find node from [nodeMapByCreatedAt]
+     * @param executedAt The TimeTicket object to remove from node.
+     * @return CrdtElement if the object exist in [elementQueueMapByKey] by same key, it will return.
+     * @throws IllegalStateException if RHTPQMapNode doesn't exist.
+     */
+    fun delete(createdAt: TimeTicket, executedAt: TimeTicket): CrdtElement {
+        val createdAtId = createdAt.toIDString()
+        val nodeMap = nodeMapByCreatedAt
+        if (!nodeMap.contains(createdAtId)) {
+            Log.e(logTag, "fail to find $createdAtId")
+        }
+
+        val node = nodeMap[createdAtId]
+            ?: throw IllegalStateException("The RHTPQMapNode by $createdAtId doesn't exist.")
+        node.remove(executedAt)
+        return node.value
+    }
+
+    /**
+     * `keyOf` returns a key of node based on creation time
+     *
+     * @param createdAt to find node from [nodeMapByCreatedAt]
+     * @return RHTPQMapNode's strKey
+     */
+    fun keyOf(createdAt: TimeTicket): String {
+        return nodeMapByCreatedAt[createdAt.toIDString()]?.strKey
+            ?: throw IllegalStateException("RHTPQMapNode's strKey by $createdAt doesn't exist")
+    }
+
+    /**
+     * `purge` physically purge child element.
+     *
+     * @param element to purge child element from [nodeMapByCreatedAt] and [elementQueueMapByKey]
+     */
+    fun purge(element: CrdtElement) {
+        val nodeMap = nodeMapByCreatedAt
+        val node = nodeMap[element.createdAt.toIDString()]
+
+        if (node == null) {
+            Log.e(logTag, "fail to find ${element.createdAt.toIDString()}")
+            return
+        }
+
+        val queue = elementQueueMapByKey[node.strKey]
+        if (queue == null) {
+            Log.e(logTag, "fail to find queue of ${node.strKey}")
```

Comment:
@hackerwins ,
Please check the original log message : https://github.com/yorkie-team/yorkie-js-sdk/blob/main/src/document/crdt/rht_pq_map.ts#L156

Reply:
Thanks for the report. I'll fix it.